cpp/src/barretenberg/dsl/acir_proofs/acir_proofs.cpp

```diff
@@ -296,19 +327,30 @@ size_t verify_recursive_proof(uint8_t const* proof_buf,
 
     acir_format::Composer composer;
 
-    transcript::Manifest manifest = acir_format::Composer::create_unrolled_manifest(1);
+    transcript::Manifest manifest = acir_format::Composer::create_unrolled_manifest(num_public_inputs);
     // We currently only support RecursionConstraint where inner_proof_contains_recursive_proof = false.
     // We would either need a separate ACIR opcode where inner_proof_contains_recursive_proof = true,
     // or we need non-witness data to be provided as metadata in the ACIR opcode
-    std::shared_ptr<acir_format::verification_key_ct> vkey =
-        acir_format::verification_key_ct::template from_field_pt_vector<false>(&composer, key_fields);
+    // recursively verify the proof
+    std::array<uint32_t, acir_format::RecursionConstraint::AGGREGATION_OBJECT_SIZE> nested_aggregation_object = {};
```

Comment:
This method could be quite slow as we are building a circuit that technically we do not need to (if we made some tweaks).

One alternative to this would be to enable the native verification algorithm to output an aggregation object. i.e. the native verifier performs a pairing check in the last step. Instead of this it would return the G1 points it inputs into the pairing; these would be the new aggregation object. 

Possibly complex as it may require also making the native verifier use the recursion-friendly hash function (not keccak). Maybe create this as an issue that we can address later?

Reply:
It does get a little complex with determining which hash function to use, but may be worth it to not generate a circuit. Is it very inefficient if we are not generating a proof? Either way I can open up an issue for this.

Reply:
Issue made here: https://github.com/AztecProtocol/barretenberg/issues/447